pkg/providers/instance/instance.go

```diff
@@ -174,19 +174,22 @@ func (p *DefaultProvider) List(ctx context.Context) ([]*Instance, error) {
 }
 
 func (p *DefaultProvider) Delete(ctx context.Context, id string) error {
-	if _, err := p.ec2Batcher.TerminateInstances(ctx, &ec2.TerminateInstancesInput{
-		InstanceIds: []string{id},
-	}); err != nil {
-		if awserrors.IsNotFound(err) {
-			return cloudprovider.NewNodeClaimNotFoundError(fmt.Errorf("instance already terminated"))
+	out, err := p.Get(ctx, id)
+	if err != nil {
+		if cloudprovider.IsNodeClaimNotFoundError(err) {
+			return err
 		}
-		if _, e := p.Get(ctx, id); e != nil {
-			if cloudprovider.IsNodeClaimNotFoundError(e) {
-				return e
+		return fmt.Errorf("terminating instance describe, %w", err)
+	}
+	if out.State != ec2types.InstanceStateNameShuttingDown {
```

Comment:
Might be nice to add a comment here that indicates why we are doing this kind of gating -- reducing the number of Delete calls that we make and reducing our overall QPS -- maybe add links to the documentation as well and then also something that notes-down that we will always eventually cleanup instances through our GC